modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleSettingLocalServiceImpl.java

```diff
@@ -53,6 +65,198 @@ public ObjectValidationRuleSetting addObjectValidationRuleSetting(
 			objectValidationRuleSetting);
 	}
 
+	@Override
+	public List<ObjectValidationRuleSetting> updateObjectValidationRuleSettings(
+			ObjectValidationRule objectValidationRule,
+			List<ObjectValidationRuleSetting> objectValidationRuleSettings)
+		throws PortalException {
+
+		List<ObjectValidationRuleSetting> keyObjectValidationRuleSettings =
+			new ArrayList<>();
+
+		List<ObjectValidationRuleSetting> outputObjectValidationRuleSettings =
+			new ArrayList<>();
+
+		_getDeleteObjectValidationRuleSettings(
+			keyObjectValidationRuleSettings, objectValidationRuleSettings,
+			outputObjectValidationRuleSettings);
+
+		long userId = objectValidationRule.getUserId();
+
+		_validate(
+			objectValidationRule, userId, keyObjectValidationRuleSettings,
+			outputObjectValidationRuleSettings);
+
+		long objectValidationRuleId =
+			objectValidationRule.getObjectValidationRuleId();
+
+		List<ObjectValidationRuleSetting> oldKeyObjectValidationRuleSettings =
+			objectValidationRuleSettingPersistence.findByOVRI_N(
+				objectValidationRuleId,
+				ObjectValidationRuleSettingConstants.NAME_KEY_OBJECT_FIELD_ID);
+
+		_deleteOldObjectValidationRuleSettings(
+			objectValidationRuleId,
+			ObjectValidationRuleSettingConstants.NAME_KEY_OBJECT_FIELD_ID,
+			keyObjectValidationRuleSettings,
+			oldKeyObjectValidationRuleSettings);
```

Comment:
Inline oldKeyObjectValidationRuleSettings and oldOutputObjectValidationRuleSettings